libraries/AP_Networking/AP_Networking_port.cpp

```diff
@@ -319,10 +319,14 @@ bool AP_Networking::Port::send_receive(void)
 {
 
     bool active = false;
-    WITH_SEMAPHORE(sem);
+    uint32_t space;
+
 
     // handle incoming packets
-    const auto space = readbuffer->space();
+    {
+        WITH_SEMAPHORE(sem);
```

Comment:
Since we're acting on different memory buckets, perhaps we should be splitting this into two semaphore: sem_write and sem_read

Reply:
we could, but it won't really gain us anything